specification/purview/data-plane/Azure.Analytics.Purview.Share/preview/2023-05-30-preview/share.json

```diff
@@ -0,0 +1,2212 @@
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SentShares_DeleteInvitation": {
+            "$ref": "./examples/SentShares_DeleteInvitation.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/sentShares/{sentShareId}/sentShareInvitations/{sentShareInvitationId}:notify": {
+      "post": {
+        "tags": [
+          "SentShareInvitation"
+        ],
+        "summary": "Notifies the recipient of the sent share invitation.",
+        "description": "Notifies the user recipient of the sent share invitation, does not apply to service invitations.",
+        "operationId": "SentShares_NotifyUserInvitation",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/sentShareId"
+          },
+          {
+            "$ref": "#/parameters/sentShareInvitationId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "in": "header",
+            "name": "repeatability-request-id",
+            "description": "If specified, the client directs that the request is repeatable; that is, that the client can make the request multiple times with the same Repeatability-Request-Id and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-Id is an opaque string representing a client-generated, globally unique for all time, identifier for the request. It is recommended to use version 4 (random) UUIDs.",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/SentShareInvitation"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/PurviewShareError"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SentShares_NotifyUserInvitation": {
+            "$ref": "./examples/SentShares_NotifyUserInvitation.json"
+          }
+        }
+      }
+    },
+    "/shareResources": {
+      "get": {
+        "tags": [
+          "ShareResource"
+        ],
+        "summary": "API operation to list ShareResources.",
+        "description": "List share resources",
+        "operationId": "ShareResources_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/skipToken"
+          },
```

Comment:
Should we avoid skipToken in parameter, as it should be handled by nextLink.

You have it in existing APIs, so we may ignore it on consistency basis.

Reply:
Hey Weidong, here is an example value of next link:
"https://accountName.purview.azure.com/share/shareResources?api-version=2023-05-30-preview&**skipToken=skip_token_to_fetch_more_share_resources**"

Here's the flow. The user makes a call to our API with no skip token passed as a parameter. We return a nextLink URL. When the user makes another API call using the next link URL, it contains a skip token this time. We use that skip token value and pass it to the db to return the next set of results. Hopefully this made sense!

Reply:
Yes, I know skipToken is used for pagination. What I mean is that the mechanism of pagination is usually not concerned by the customer, if "nextLink" is used. Customer or SDK will just send the URL of "nextLink" for the next page. And the "skipToken" more or less become an implementation detail (and customer are not able to make a skipToken by themselves).

But I do see some swagger did include the "skipToken" in parameter. So it is up to you to decide what to do with it. Anyway it is a preview.

But if it is a GA, you are no longer able to remove the "skipToken", even if you change your pagination implementation.

Reply:
I see, so the suggestion is to remove it from the specs file? But still have the back-end API support it as a parameter.